test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -145,6 +145,10 @@ def test_sim_recursive_h_wrapper():
 
 
 def test_recursive_h():
+    @aq.main(num_qubits=6)
+    def main(n: int):
+        recursive_h(n)
```

Comment:
Interesting, I guess we will not be able to support recursive `main` functions. Probably for the best, that doesn't make much sense anyway and it's very easy to wrap it as you have done here.

Reply:
Agree. It feels like we might save our users from some weird bugs by disallowing this